pkg/generate/app/pipeline.go

```diff
@@ -180,43 +181,62 @@ func (s sortablePorts) Swap(i, j int) {
 	s[j] = p
 }
 
+// AddServices sets up services for the provided objects
 func AddServices(objects Objects) Objects {
 	svcs := []runtime.Object{}
 	for _, o := range objects {
 		switch t := o.(type) {
 		case *deploy.DeploymentConfig:
 			for _, container := range t.Template.ControllerTemplate.Template.Spec.Containers {
+				name, generateName := makeValidServiceName(t.Name)
+				svc := &kapi.Service{
+					ObjectMeta: kapi.ObjectMeta{
+						Name:         name,
```

Comment:
@smarterclayton every service gets its name from the name of the deploymentConfig. But what if that deployment specifies more than one container in a pod? As is, say there is a pod with three containers, we will get three services with the same name.